src/main/scala/utils/images/Image.scala

```diff
@@ -251,8 +274,36 @@ trait VectorizedImage extends Image {
   override def put(x: Int, y: Int, channelIdx: Int, newVal: Double) = {
     putInVector(imageToVectorCoords(x, y, channelIdx), newVal)
   }
+
+  def vectorToImageCoords(v: Int): Coordinate = ???
+
+  val coord: Coordinate = new Coordinate(0,0,0)
+
+  def iter(): Iterator[CoordinateValue] = new Iterator[CoordinateValue] {
```

Comment:
Also calling iter from multiple threads will cause bad things to happen from what i can see as `coord` is updated by all the calls to `vectorToImageCoords` -- is that right ?